### PR TITLE
Update TiDB Config to v3.1.0

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve? 
Update TidbConfig for TidbCluster to integrated with tidb 3.1

ref https://github.com/pingcap/tidb-operator/issues/1662

Related changes

 - Need to cherry-pick to the release branch

### Does this PR introduce a user-facing change?:

```release-note
Update TidbConfig for TidbCluster to tidb v3.1.0
```

### CODE REVIEW

File: pkg/apis/pingcap/v1alpha1/tidb_config.go [hunk 1]
```diff
@@ -103,6 +103,27 @@ type TiDBConfig struct {
...
+	// NewCollationsEnabledOnFirstBootstrap indicates if the new collations are enabled, it effects only when a TiDB cluster bootstrapped on the first time.
+	// +optional
+	NewCollationsEnabledOnFirstBootstrap bool `toml:"new_collations_enabled_on_first_bootstrap" json:"new_collations_enabled_on_first_bootstrap,omitempty"`
+	// Experimental contains parameters for experimental features.
+	// +optional
+	Experimental Experimental `toml:"experimental" json:"experimental,omitempty"`
+	// EnableDynamicConfig enables the TiDB to fetch configs from PD and update itself during runtime.
+	// see https://github.com/pingcap/tidb/pull/13660 for more details.
+	// +optional
+	EnableDynamicConfig bool `toml:"enable-dynamic-config" json:"enable-dynamic-config,omitempty"`
```
Comment (cofyc) [inline pkg/apis/pingcap/v1alpha1/tidb_config.go]: add tidb version requirements, links to configuration docs in comments? these comments can be our docs for API types.

Comment (cofyc) [reply pkg/apis/pingcap/v1alpha1/tidb_config.go]: we can generate API docs for our CR types like this: https://cert-manager.io/docs/reference/api-docs/ .

Comment (Yisaer) [reply pkg/apis/pingcap/v1alpha1/tidb_config.go]: good catch

Comment (Yisaer) [reply pkg/apis/pingcap/v1alpha1/tidb_config.go]: Currently docs team is now working on marked big version for the configuration. Do we need to pend this issue or we merge this pr and mark the big version unit their work is done?

Comment (Yisaer) [reply pkg/apis/pingcap/v1alpha1/tidb_config.go]: After discussion from docs team, I think we could pend this issue and wait their version makring work done.

Comment (Yisaer) [reply pkg/apis/pingcap/v1alpha1/tidb_config.go]: currently the newly 3.1.0 added configuration is makred in note.

**Comment by cofyc** [comment]
remove WIP prefix and label when it's ready to review

**Pull request merged**